### ISSUE TITLE
Improved feedback when a curseforge project is considered incompatible

### ISSUE THREAD

**Issue by dashkal16** [theRookieCoder/ferium#76]
At present, when a curseforge project is not considered compatible, that's all the user is told:
```
PS D:\Games\PolyMC\instances\BananaCraft_1.18.2\.minecraft\mods> ferium add-curseforge 575698
Adding mod... The project/repository is not compatible
```

Could the error message be expanded to explain what filter(s) were not met among the files in the project?
In the above case, it turned out to be something governed by `--dont-check-game-version` but that was not at all obvious, especially as the mod in question does declare that it works for 1.18.2

```
PS D:\Games\PolyMC\instances\BananaCraft_1.18.2\.minecraft\mods> ferium add-curseforge --dont-check-game-version 575698
Adding mod... ✓ (Ars Creo)
Adding required dependency... ✓ Already added
Adding required dependency... ✓ Already added
```

As of this writing, the newest file for Ars Creo is https://www.curseforge.com/minecraft/mc-mods/ars-creo/files/3776248
Which is flagged for `1.18.2` and `Forge`

**Comment by KyleUltimate** [comment]
smth like
`Adding mod... The project/repository is not compatible with your game version.`
```add `--dont-check-game-version` if you insist to add it.```
and for mod loader errors
`Adding mod... The project/repository is not compatible with your mod loader.`
```add `--dont-check-mod-loader` if you insist  to add it.```

**Comment by theRookieCoder** [comment]
The problem currently is that the checks are done by a single for loop and if statement (e.g [curseforge checking](https://github.com/theRookieCoder/libium/blob/main/src/upgrade/check.rs#L30-L39)) so this would probably require a completely different method to be used

**Comment by theRookieCoder** [comment]
And yeah for some reason Ars Creo is not adding even with the proper game version and mod loader

**Comment by aschoettler** [comment]
I also found that [`Earthcomputer/clientcommands`](https://github.com/Earthcomputer/clientcommands) and [`https://github.com/Coow/Cows-Odd-Widgets`](https://github.com/Coow/Cows-Odd-Widgets) failed to download for 1.18.2 (even with `--dont-check-modloader`) despite the releases saying they are for 1.18.2.

**Comment by theRookieCoder** [comment]
Yes because the version is not mentioned in the actual asset name. **Sigh** we really have to overhaul the GitHub releases checking

**Comment by dashkal16** [comment]
This hit my radar again so I checked the API myself.
The file in question doesn't have version tags when queried through the API.
They show fine in the web UI, but not over the API.  So that's fun.

I checked all three of
https://api.curseforge.com/v1/mods/575698
https://api.curseforge.com/v1/mods/575698/files
https://api.curseforge.com/v1/mods/575698/files/3776248

With my own API key and the file's there, but `"gameVersions": []`
But...
![image](https://user-images.githubusercontent.com/464894/171532557-fa90deb1-251f-4a69-93b4-3a32564f627d.png)

**Comment by theRookieCoder** [comment]
I don't know how the CF website works, but it probably uses a different internal API or somehow cached responses? This is an issue with CF, and I've contacted them on the discord CF Dev server and they just didn't reply at all. There's nothing I can do about this :(

I suggest you download the mod and place it in the user folder for noe

**Comment by dashkal16** [comment]
Indeed not.  Hence the actual request.
Being told which flags would have allowed the command to succeed provides a work-around.
